pingcap/tidb
# executor: fix log desensitization bug in prestmt

Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Close https://github.com/pingcap/tidb/issues/18566#issuecomment-677724152

When tidb log enables desensitization  (`set @@global.tidb_log_desensitization = 1;`), the slow-log and log may still out put the previous statement that doesn't desensitization:


**before**
```sql
...
# Prev_stmt: insert into t values (1,1),(2,2),(3,3),(4,4),(5,5)
commit;
```

**this PR**

```sql
...
# Prev_stmt: insert into t values ( ... ) , ( ... ) , ( ... ) , ( ... ) , ( ... )
commit;

```


### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test

```sql
set @@tidb_slow_log_threshold=0;
begin;insert into t values (1,1),(2,2),(3,3),(4,4),(5,5); 
commit;
set @@tidb_slow_log_threshold=100;
-- Then check the tidb-slow-log file.
```

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

- Fix log desensitization bug in prestmt
